container/Dockerfile.vllm

```diff
@@ -179,7 +179,11 @@ RUN uv pip install /workspace/wheels/nixl/*.whl
 
 # Install vllm - keep this early in Dockerfile to avoid
 # rebuilds from unrelated source code changes
-ARG VLLM_REF="059d4cd"
+# [gluo NOTE] currently using a fork of vllm until the fix
+# for multi-modal disaggregated serving is merged upstream.
+# see https://github.com/vllm-project/vllm/pull/21074
+ARG VLLM_REPO=https://github.com/GuanLuo/vllm.git
+ARG VLLM_REF="eaadf838ebe93e29a38a6fc1bab5a9801abe7d2c"
```

Comment:
CC @alec-flowers @ptarasiewiczNV @nv-tusharma  who have also been modifying vllm versions/build - to make sure there's no thrashing of each other's work here